src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -511,22 +513,24 @@ void InitializeCaches()
                 _configCache = ((IBuildComponentHost)this).GetComponent(BuildComponentType.ConfigCache) as IConfigCache;
                 _resultsCache = ((IBuildComponentHost)this).GetComponent(BuildComponentType.ResultsCache) as IResultsCache;
 
-                if (!usesInputCaches && (_buildParameters.ResetCaches || _configCache.IsConfigCacheSizeLargerThanThreshold()))
+                if (usesInputCaches)
```

Comment:
Affects non-isolation execution: just a condition simplification, should be benign.